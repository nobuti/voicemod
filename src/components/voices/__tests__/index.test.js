import React from "react";
import { render, screen } from "@testing-library/react";
import Voices from "../";

describe("Voices", () => {
  test("should render properly", () => {
    const props = {
      voices: [
        { id: "wadus", name: "Wadus", icon: "VoicesVoiceIcon01.png" },
        { id: "manzana", name: "Manzana", icon: "VoicesVoiceIcon02.png" },
        { id: "cebolleta", name: "Cebolleta", icon: "VoicesVoiceIcon01.png" },
      ],
      favorites: ["manzana", "cebolleta"],
    };

    const { container } = render(<Voices {...props} />);
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId("grid-title").textContent).toEqual("Pro voices");
  });

  test("should pass favorite prop properly", () => {
    const props = {
      voices: [
        { id: "manzana", name: "Manzana", icon: "VoicesVoiceIcon02.png" },
      ],
      favorites: ["manzana"],
    };

    render(<Voices {...props} />);
    expect(screen.queryByTestId("voice")).toHaveClass("is-favorite");
  });

  test("should show empty message properly", () => {
    const props = {
      voices: [],
    };

    render(<Voices {...props} />);
    expect(screen.getByTestId("voices-empty")).toBeInTheDocument();
  });
});
