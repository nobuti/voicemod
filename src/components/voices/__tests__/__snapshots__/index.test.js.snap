// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Voices should render properly 1`] = `
<div>
  <section
    class="Section"
    data-testid="grid"
  >
    <h1
      class="Heading"
      data-testid="grid-title"
    >
      Pro voices
    </h1>
    <ul
      class="Grid"
      data-testid="grid-list"
    >
      <li>
        <div
          class="Voice"
          data-testid="voice"
          tabindex="0"
        >
          <div
            class="Voice-image"
          >
            <img
              alt="Wadus"
              src="/VoicesVoiceIcon01.png"
            />
            ;
          </div>
          <div
            class="Voice-name"
          >
            Wadus
          </div>
          <button
            class="Voice-favorite"
            data-testid="voice-fav"
          >
            Fav me
          </button>
        </div>
      </li>
      <li>
        <div
          class="Voice is-favorite"
          data-testid="voice"
          tabindex="0"
        >
          <div
            class="Voice-image"
          >
            <img
              alt="Manzana"
              src="/VoicesVoiceIcon02.png"
            />
            ;
          </div>
          <div
            class="Voice-name"
          >
            Manzana
          </div>
          <button
            class="Voice-favorite"
            data-testid="voice-fav"
          >
            Please don't
          </button>
        </div>
      </li>
      <li>
        <div
          class="Voice is-favorite"
          data-testid="voice"
          tabindex="0"
        >
          <div
            class="Voice-image"
          >
            <img
              alt="Cebolleta"
              src="/VoicesVoiceIcon01.png"
            />
            ;
          </div>
          <div
            class="Voice-name"
          >
            Cebolleta
          </div>
          <button
            class="Voice-favorite"
            data-testid="voice-fav"
          >
            Please don't
          </button>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
